jakttest/jakttest.jakt

```diff
@@ -59,13 +63,96 @@ function strip_line_breaks(anon input: String) throws -> String {
     return builder.to_string()
 }
 
+function strip_prefix(anon value: String, prefix: String) throws -> String? {
+    let no_value: String? = None
+    if value.length() < prefix.length() {
+        return no_value
+    }
+
+    for i in 0..prefix.length() {
+        if value.byte_at(i) != prefix.byte_at(i) {
+            return no_value
+        }
+    }
+
+    return value.substring(start: prefix.length()
+                          length: value.length() - prefix.length())
+
+}
+
+
+struct Options {
+    file_name: String
+    temp_dir: String
+    errors: [String]
+
+    function from_args(args: [String]) throws -> Options {
+        mut options = Options(file_name: "", temp_dir: ".", errors: [])
+
+        mut index = 1uz
+        while index != args.size() {
+            if args[index] == "-T" {
+                index++
+                if index == args.size() {
+                    options.errors.push("-T given without a value")
+                    break // we break since we met our looping condition
+                }
+                options.temp_dir = args[index]
+                index++
+                continue
+            }
+
+            {
+                // FIXME: when better Optional support arrives this should be
+                // a match.
+                let temp_dir = strip_prefix(args[index], prefix: "--temp-dir")
+                if temp_dir.has_value() {
+                    if temp_dir.value().byte_at(0) != b'=' {
+                        options.errors.push("--temp-dir given without a value")
+                        // skip it so the rest of arguments can be validated
+                        index++
+                        continue
+                    }
+
+                    options.temp_dir = temp_dir!.substring(start: 1, length: temp_dir!.length() - 1)
```

Comment:
maybe we could check if the dir name starts with `"` or `'` and remove those too?

Reply:
Getting the quote to be inside the dirname from the shell arguments is actually difficult to pull off. When you use `'` or `"` in a shell argument what you mean is that the inside of the quotes should be a single argument: `git commit -m 'hello world'` makes `hello world` the third argument

Reply:
So when you do `jakttest --temp-dir="hello world" sample.jakt`, `args` is: `["jakttest", "--temp-dir=hello world", "sample.jakt"]`.

Reply:
Oh yeah, true. :^)